src/org/infoglue/calendar/controllers/EventController.java

```diff
@@ -1993,24 +1994,58 @@ public boolean hasUserGroup(InfoGluePrincipalBean principal, Event event) throws
         
         return false;
     }
-    
-    public List getAssetKeys()
-    {
-        List assetKeys = new ArrayList();
-        
-        int i = 0;
-        String assetKey = PropertyHelper.getProperty("assetKey." + i);
-        while(assetKey != null && assetKey.length() > 0)
-        {
-            assetKeys.add(assetKey);
-            
-            i++;
-            assetKey = PropertyHelper.getProperty("assetKey." + i);
-        }
-        
-        return assetKeys;
-    }
 
+	/**
+	 * Calls {@linkplain #getAssetKeys(Event)} with null as Event.
+	 * @return
+	 */
+	public List getAssetKeys()
+	{
+		return getAssetKeys(null);
+	}
+
+	private boolean isUndefinedAssetKey(String assetKey)
+	{
+		return assetKey.startsWith("@assetKey") && assetKey.endsWith("@");
+	}
+
+	/**
+	 * Gets all asset keys defined in the system filtered based on the asset keys already used by the given event.
+	 * @param event
+	 * @return
+	 */
+	public List<String> getAssetKeys(Event event)
+	{
+		List<String> assetKeys = new ArrayList<String>();
+
+		int i = 0;
+		String assetKey = PropertyHelper.getProperty("assetKey." + i);
+		while (assetKey != null && assetKey.length() > 0)
+		{
+			assetKeys.add(assetKey);
+
+			i++;
+			assetKey = PropertyHelper.getProperty("assetKey." + i);
+			if (isUndefinedAssetKey(assetKey))
+			{
+				assetKey = null;
+			}
+		}
+
```

Comment:
In the code above, it looks like that assetKey.i will be added even if it matched '@assetKey...@', and then the loop will end. Should the check for isUndefinedAssetKey be done before adding the assetKey?